File: capture-sdk/sdk/src/androidTest/java/net/gini/android/capture/test/PhotoSubject.java
package net.gini.android.capture.test;

import com.google.common.truth.FailureMetadata;
import com.google.common.truth.Subject;

import net.gini.android.capture.internal.camera.photo.JpegByteArraySubject;
import net.gini.android.capture.internal.camera.photo.Photo;

import androidx.annotation.Nullable;

public class PhotoSubject extends Subject {

    private final JpegByteArraySubject mJpegByteArraySubject;

    public static Factory<PhotoSubject, Photo> photo() {
        return new Factory<PhotoSubject, Photo>() {

            @Override
            public PhotoSubject createSubject(FailureMetadata metadata, Photo actual) {
                return new PhotoSubject(metadata, actual);
            }
        };
    }

    private final Photo actual;

    protected PhotoSubject(FailureMetadata metadata, @org.checkerframework.checker.nullness.qual.Nullable Object actual) {
        super(metadata, actual);
        isNotNull();
        this.actual = (Photo) actual;
        //noinspection ConstantConditions
        mJpegByteArraySubject = new JpegByteArraySubject(metadata, this.actual.getData());
    }

    public void hasContentIdInUserComment(final String contentId) {
        final String key = "User Comment ContentId";

        if (contentId == null) {
            failWithActual(key, null);
            return;
        }

        mJpegByteArraySubject.hasContentIdInUserComment(contentId);
    }

    public void hasRotationDeltaInUserComment(final int rotationDelta) {
        mJpegByteArraySubject.hasRotationDeltaInUserComment(rotationDelta);
    }

    public void hasSameUserCommentAs(@Nullable final Photo photo) {
        final String key = "User Comment";
        if (photo == null) {
            failWithActual(key, null);
            return;
        }
        mJpegByteArraySubject.hasSameUserCommentAs(photo.getData());
    }
}

File: capture-sdk/sdk/src/test/java/net/gini/android/capture/test/PhotoSubject.kt
package net.gini.android.capture.test

import com.google.common.truth.FailureMetadata
import com.google.common.truth.Subject
import com.google.common.truth.Subject.Factory
import net.gini.android.capture.internal.camera.photo.JpegByteArraySubject
import net.gini.android.capture.internal.camera.photo.Photo

class PhotoSubject constructor(metadata: FailureMetadata?, actual: Any?) : Subject(metadata, actual) {
    private val mJpegByteArraySubject: JpegByteArraySubject
    private val actual: Photo?

    init {
        isNotNull()
        this.actual = actual as Photo?
        mJpegByteArraySubject = JpegByteArraySubject(metadata, this.actual!!.data)
    }

    fun hasContentIdInUserComment(contentId: String?) {
        val key = "User Comment ContentId"
        if (contentId == null) {
            failWithActual(key, null)
            return
        }
        mJpegByteArraySubject.hasContentIdInUserComment(contentId)
    }

    fun hasRotationDeltaInUserComment(rotationDelta: Int) {
        mJpegByteArraySubject.hasRotationDeltaInUserComment(rotationDelta)
    }

    fun hasSameUserCommentAs(photo: Photo?) {
        val key = "User Comment"
        if (photo == null) {
            failWithActual(key, null)
            return
        }
        mJpegByteArraySubject.hasSameUserCommentAs(photo.data)
    }

    companion object {
        fun photo(): Factory<PhotoSubject, Photo> {
            return Factory { metadata, actual -> PhotoSubject(metadata, actual) }
        }
    }
}